### PR TITLE
Per-department wire layouts

### PR DESCRIPTION
**Pull request by Killerqu00**
## About the PR
Wire layouts for airlocks are now only same inside a single department. Wire layouts for different department airlocks are different.
 ~~Also, airlocks now generally have more wires.~~

## Why / Balance

See Discord discussions and feedback regarding MRP hacking clarifications:
https://ptb.discord.com/channels/310555209753690112/1217557314328854628/1217779616828231691

## Technical details

Made some WiresSystem.cs changes, fixing code that broke inheritance.

## Media

image
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Airlock wires are now different between departments.

### CODE REVIEW

**Comment by Emisse** [comment]
thank you my genius child

**Comment by asperger-sind** [comment]
I really don't think all departments should have as much wires as a vault door, especially if the wireset isn't the same for every department. Hacking is already locked behind functional insuls and tools, now with airlock plating introduced I don't really see a reason to make hacking even worse by increasing the number of wires you need to go through before the door is open.

**Comment by Killerqu00** [comment]
> I really don't think all departments should have as much wires as a vault door, especially if the wireset isn't the same for every department. Hacking is already locked behind functional insuls and tools, now with airlock plating introduced I don't really see a reason to make hacking even worse by increasing the number of wires you need to go through before the door is open.

I will be honest - for some reason, this is just what happens when inheriting wire layouts. It just adds more dummy wires. Why? No idea.

**Comment by LordEclipse** [comment]
I like redundancy wires! I also prefer every department airlock having unique wiresets because otherwise its:
>Find random airlock
>Find all the proper wires
>You now know how to open every door on the station except Sec and Command

However, there could be more utility wires such as AI/Borg wire that would prevent borgs from opening doors, just to name an example.

**Comment by Emisse** [comment]
i think this is fine

**Comment by Plykiya** [comment]
Dummy wires are neat.

**Comment by Killerqu00** [comment]
> I like redundancy wires! I also prefer every department airlock having unique wiresets because otherwise its:
> 
> > Find random airlock
> > Find all the proper wires
> > You now know how to open every door on the station except Sec and Command
> 
> However, there could be more utility wires such as AI/Borg wire that would prevent borgs from opening doors, just to name an example.

I think some maintainer requested to port a system from TG completely eventually.

**Comment by K-Dynamic** [comment]
> I think some maintainer requested to port a system from TG completely eventually.

Just turn the ui 90 degrees :trollface:

**Comment by asperger-sind** [comment]
Randomized wiresets are fine, perhaps even good, but dummy wires just aren't well-fit for the hacking balance situation right now imo.

**Review by VasilisThePikachu — COMMENTED** [review]
I feel like 12 wires are a little excessive. Potentially just add some dummy wires instead?

**Comment by Tayrtahn** [comment]
I recommend submitting the wire inheritance thing as an issue. Maybe it can be made optional or something.

**Review by PJB3005 — DISMISSED** [review]
> I will be honest - for some reason, this is just what happens when inheriting wire layouts. It just adds more dummy wires. Why? No idea.

Fix the dummy wires thing. Relying on the bug is just gonna cause trouble later on.

**Comment by Carmell0** [comment]
Can we PLEASE change the sala rule now

File: Resources/Prototypes/Wires/layouts.yml [hunk 1]
```diff
@@ -10,9 +10,29 @@
...
+  parent: Airlock
+  id: AirlockEngineering
+
+- type: wireLayout
+  parent: Airlock
+  id: AirlockMedical
+
+- type: wireLayout
+  parent: HighSec
+  id: AirlockScience
```
Comment (VasilisThePikachu) [inline Resources/Prototypes/Wires/layouts.yml:34]: Why does science have high-sec doors?

File: Resources/Prototypes/Wires/layouts.yml [hunk 2]
```diff
@@ -10,9 +10,29 @@
   - !type:DoorSafetyWireAction
 
 - type: wireLayout
-  parent: Airlock
+  parent: HighSec
   id: AirlockSecurity
```
Comment (VasilisThePikachu) [inline Resources/Prototypes/Wires/layouts.yml:14]: Think Security should keep regular airlock wires.

**Review by VasilisThePikachu — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Wires/layouts.yml (hunk 1 above)
Comment (Killerqu00) [reply Resources/Prototypes/Wires/layouts.yml]: my bad, probably misread science as security

File: Resources/Prototypes/Wires/layouts.yml (hunk 2 above)
Comment (Killerqu00) [reply Resources/Prototypes/Wires/layouts.yml]: fixed

**Comment by Killerqu00** [comment]
> > I will be honest - for some reason, this is just what happens when inheriting wire layouts. It just adds more dummy wires. Why? No idea.
> 
> Fix the dummy wires thing. Relying on the bug is just gonna cause trouble later on.

Done - apparently layoutPrototype already contains wires from parents for some reason. LMK if my solution is too hacky.

**Review by VasilisThePikachu — APPROVED** [review]
lgtm

**Comment by PJB3005** [comment]
I made a proper fix for the wire layout inheritance issue: https://github.com/space-wizards/space-station-14/pull/26289

**Comment by VasilisThePikachu** [comment]
> I made a proper fix for the wire layout inheritance issue: #26289

Merged, revert the code changes here as they are no longer needed and we are good

**Pull request merged**